test/system/src/test/java/io/pravega/test/system/ConsumptionBasedRetentionWithMultipleReaderGroupsTest.java

```diff
@@ -0,0 +1,287 @@
+        ExecutorServiceHelpers.shutdown(streamCutExecutor);
+    }
+
+    @Test
+    public void multipleSubscriberCBRTest() throws Exception {
+        final ClientConfig clientConfig = Utils.buildClientConfig(controllerURI);
+
+        @Cleanup
+        EventStreamClientFactory clientFactory = EventStreamClientFactory.withScope(SCOPE, clientConfig);
+        @Cleanup
+        EventStreamWriter<String> writer = clientFactory.createEventWriter(STREAM, new JavaSerializer<>(),
+                EventWriterConfig.builder().build());
+
+        // Write 7 events to the stream.
+        for (int i = 1; i <= 7; i++) {
+            log.info("Writing event e{} to {}/{}", i, SCOPE, STREAM);
+            writer.writeEvent(SIZE_30_EVENT).join();
+        }
+
+        @Cleanup
+        ReaderGroupManager readerGroupManager = ReaderGroupManager.withScope(SCOPE, clientConfig);
+        boolean createRG1 = readerGroupManager.createReaderGroup(READER_GROUP_1, ReaderGroupConfig.builder()
+                .retentionType(ReaderGroupConfig.StreamDataRetention.MANUAL_RELEASE_AT_USER_STREAMCUT)
+                .disableAutomaticCheckpoints()
+                .stream(Stream.of(SCOPE, STREAM)).build());
+        assertTrue("Reader group 1 is not created", createRG1);
+        boolean createRG2 = readerGroupManager.createReaderGroup(READER_GROUP_2, ReaderGroupConfig.builder()
+                .retentionType(ReaderGroupConfig.StreamDataRetention.MANUAL_RELEASE_AT_USER_STREAMCUT)
+                .disableAutomaticCheckpoints()
+                .stream(Stream.of(SCOPE, STREAM)).build());
+        assertTrue("Reader group 2 is not created", createRG2);
+
+        ReaderGroup readerGroup1 = readerGroupManager.getReaderGroup(READER_GROUP_1);
+        ReaderGroup readerGroup2 = readerGroupManager.getReaderGroup(READER_GROUP_2);
+        @Cleanup
+        EventStreamReader<String> reader1 = clientFactory.createReader(READER_GROUP_1 + "-" + 1,
+                READER_GROUP_1, new JavaSerializer<>(), readerConfig);
+        @Cleanup
+        EventStreamReader<String> reader2 = clientFactory.createReader(READER_GROUP_2 + "-" + 1,
+                READER_GROUP_2, new JavaSerializer<>(), readerConfig);
+
+        EventRead<String> read;
+        EventRead<String> read2;
+        // Read three events with reader1.
+        for (int i = 1; i <= 3; i++) {
+            read = reader1.readNextEvent(READ_TIMEOUT);
+            assertEquals("data of size 30", read.getEvent());
+        }
+        log.info("{} generating 1st stream-cuts for {}/{}", READER_GROUP_1, SCOPE, STREAM);
+        CompletableFuture<Map<Stream, StreamCut>> futureCuts1 = readerGroup1.generateStreamCuts(streamCutExecutor);
+        // Wait for 5 seconds to force reader group state update. This will allow for the silent
+        // checkpoint event generated as part of generateStreamCuts to be picked and processed.
+        Exceptions.handleInterrupted(() -> TimeUnit.SECONDS.sleep(5));
```

Comment:
Instead of sleep its recommended to use delayed futures.

Reply:
updated to use` Futures.delayedFuture(Duration.ofSeconds(5), executor).join();`